src/evergreen/api.py

```diff
@@ -829,7 +829,8 @@ def single_test_by_task_and_test_file(self, task_id: str, test_file: str) -> Tst
         :return: the test for the specified task.
         """
         url = self._create_url(f"/tasks/{task_id}/tests")
-        return Tst(self._call_api(url, params={"test_name": test_file}).json(), self)
+        test = self._call_api(url, params={"test_name": test_file}).json()
+        return test[0]
```

Comment:
We should not be accessing an array element without first checking its length. If we happen to get an empty list back, which seems like it could be valid, this will through an exception.

Since evergreen is returning a list for endpoint, we can return a list as well. 

I think we can just do something like this:

```python
return [Tst(t, self) for t in self._call_api(url, params=params).json()]
```